src/app/signup/page.jsx

```diff
@@ -35,22 +35,18 @@ export default function SignupPage() {
       router.push("/login");
     } catch (error) {
       console.error("Signup error:", error);
-      
+
       // User-friendly error messages
-      if (error.code === 'auth/email-already-in-use') {
-        setError('This email is already registered. Please log in instead.');
-      } 
-      else if (error.code === 'auth/invalid-email') {
-        setError('Please enter a valid email address.');
-      }
-      else if (error.code === 'auth/weak-password') {
-        setError('Password should be at least 6 characters.');
-      }
-      else if (error.code === 'auth/operation-not-allowed') {
-        setError('Email/password sign-up is currently disabled.');
-      }
-      else {
-        setError('Something went wrong. Please try again.');
+      if (error.code === "auth/email-already-in-use") {
+        setError("This email is already registered. Please log in instead.");
+      } else if (error.code === "auth/invalid-email") {
+        setError("Please enter a valid email address.");
+      } else if (error.code === "auth/weak-password") {
+        setError("Password should be at least 6 characters.");
+      } else if (error.code === "auth/operation-not-allowed") {
+        setError("Email/password sign-up is currently disabled.");
+      } else {
+        setError("Something went wrong. Please try again.");
```

Comment:
move this code to a utility function. create a utility folder at the root of your repo
```suggestion
     const errorMessage =  getUserFriendlyErrorMessage(error.code);
     setError(errorMessage);
```